### PR TITLE
feat(experimental): `Icon` add new component

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/experimental/icon/examples/import/insert-template.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```html
+<tui-icon>"Batman"</tui-icon>
```
Comment (nsbarsukov) [inline projects/demo/src/modules/experimental/icon/examples/import/insert-template.md]: Why as `ng-content` ?
Should it be `<tui-icon icon="Batman"></tui-icon>` ?

File: projects/demo/src/modules/experimental/icon/icon.template.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        >
+            <tui-icon-example-2></tui-icon-example-2>
+        </tui-doc-example>
+
+        <tui-doc-example
+            id="duo"
+            heading="Two colors"
+            [content]="example3"
+        >
+            <tui-notification class="tui-space_bottom-4">This example only works in modern browsers</tui-notification>
```
Comment (nsbarsukov) [inline projects/demo/src/modules/experimental/icon/icon.template.html:36]: Should we add link to `Can i use`?

```suggestion
            <tui-notification class="tui-space_bottom-4">This example only works in <a tuiLink href="https://caniuse.com/?search=mask-mode">modern browsers</a></tui-notification>
```

I think this example can cause many questions from our users (chrome is modern browser but it does not support this feature)?

Comment (nsbarsukov) [reply projects/demo/src/modules/experimental/icon/icon.template.html:36]: Should we hide this example if it does not support this feature ?

Something like the following:
```less
@supports not (mask-mode: luminance) {
    .all-example {
        display: none;
    }
}
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/experimental/icon/examples/import/insert-template.md (hunk 1 above)
Comment (waterplea) [reply projects/demo/src/modules/experimental/icon/examples/import/insert-template.md]: Fixed

File: projects/demo/src/modules/experimental/icon/icon.template.html (hunk 2 above)
Comment (waterplea) [reply projects/demo/src/modules/experimental/icon/icon.template.html:36]: Good idea. I believe Chrome 120 actually should support it, even though it is not mentioned on caniuse. It comes out in a month, we'll see :)

Comment (waterplea) [reply projects/demo/src/modules/experimental/icon/icon.template.html]: Yes, tested in Canary, it will work starting next major version.

**Pull request merged**